### PR TITLE
HITL - Handle episode lists instead of ranges.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

In previous iterations, the HITL server accepted episode index ranges in the format `episodes: 100-104`.

This approach had multiple issues:
* In a crowdsource context, it was not possible to "recycle" individual failed episodes. Entire ranges had to be discarded.
* If an error occurred during the middle of a session, data from the completed episodes was discarded.
* Special care was required to sequence episode sets:
    * Tutorial episodes needed to be copied every N indices.
    * Scenes required to be sequenced in order.
* Once an application was deployed, it was not possible to adjust task distribution.
* Ambiguous range boundaries causing human errors.

This changeset changes the episode format to accept ranges instead, in the format `episodes: 100,101,102,103`

## How Has This Been Tested

Tested on distributed HITL application.

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/rearrange_v2/app_state_start_session.py [hunk 1]
```diff
@@ -71,79 +79,171 @@ def _try_get_episode_ids(self) -> Optional[List[str]]:
...
+                print(
+                    f"Episode index '{episode}' is out of bounds. There are '{total_episode_count}' episodes in this dataset. Cancelling session."
+                )
+                return None
+
+        return list(episodes[0])
+
+
+if __name__ == "__main__":
+    # TODO: Temporary test. Move to a dedicated test file.
```
Comment (eundersander) [inline examples/hitl/rearrange_v2/app_state_start_session.py:144]: This is too messy to merge to `main`. You should go ahead and do this todo before merging.

File: examples/hitl/rearrange_v2/app_state_start_session.py [hunk 2]
```diff
@@ -71,79 +79,171 @@ def _try_get_episode_ids(self) -> Optional[List[str]]:
...
+                    f"Episodes are supplied in an unexpected format: '{type(episodes_str)}'. Cancelling session."
                 )
                 return None
 
-        # Validate that the episode set is not empty.
-        if episodes_str is None or len(episodes_str) == 0:
-            print("Users did not request episodes. Cancelling session.")
-            return None
+            # Parse episode string.
+            episodes_split = episodes_str.split(",")
```
Comment (eundersander) [inline examples/hitl/rearrange_v2/app_state_start_session.py:109]: Beware you are kinda duplicating the `episode_filter` logic in `hitl_driver.py`. Consider reusing that instead.

**Review by eundersander — CHANGES_REQUESTED** [review]
See comments.

File: examples/hitl/rearrange_v2/app_state_start_session.py (hunk 2 above)
Comment (0mdc) [reply examples/hitl/rearrange_v2/app_state_start_session.py:109]: This is a planned refactor post single-learn.

Lots of the code in the `rearrange_v2` can be moved to `habitat-hitl`.

**Review by eundersander — APPROVED** [review]
LGTM!

**Pull request merged**